molten-cache/src/main/java/com/hotels/molten/cache/resilience/ResilientReactiveCache.java

```diff
@@ -42,31 +44,42 @@
 
 /**
  * Resilient {@link ReactiveCache} with timeout and circuit-breaker.
- * Should be wrapped around an {@link ResilientSharedReactiveCache}.
+ * Consider using {@link ResilientReactiveCacheBuilder} to create a more robust resilient cache.<br>
+ * Should be wrapped around an {@link ResilientSharedReactiveCache}.<br>
  * Please note that the provided {@link CircuitBreakerConfig} will be altered in a way it ignores {@link BulkheadFullException} in the causal chain as a failure.
  * Instruments circuit-breaker with {@link CircuitBreakerInstrumenter}.
  *
  * @param <K> the type of key
  * @param <V> the type of value
  * @see ResilientSharedReactiveCache
+ * @see ResilientReactiveCacheBuilder
  */
-@SuppressWarnings("unchecked")
 public class ResilientReactiveCache<K, V> implements ReactiveCache<K, V> {
     private static final String HIERARCHICAL_METRICS_PREFIX = "reactive-cache";
-    private static final Map<String, String> USED_CACHE_NAMES = new ConcurrentHashMap<>();
+    private static final Set<String> USED_CACHE_NAMES = ConcurrentHashMap.newKeySet();
```

Comment:
I was thinking if this should come from a common place for both `ResilientReactiveCache`, `ResilientSharedReactiveCache`, and other future `ReactiveCache` implementations where wanted. What do you think?

Reply:
Whatever, I'll do that in a separate PR. 🙂

Reply:
Well, you could create something like a cache name registry. Or just an abstraction over this construct to simplify its reuse.